auth/api/iam/api.go

```diff
@@ -467,20 +478,8 @@ func (r Wrapper) handleAuthorizeRequest(ctx context.Context, ownDID did.DID, own
 		// TODO: officially flow switching has to be determined by the client_id
 		// registered client_ids should list which flow they support
 		// client registration could be done via rfc7591....
-		// for now we switch on client_id format.
-		// when client_id is a did:web, it is a cloud/server wallet
-		// otherwise it's a normal registered client which we do not support yet
-		// Note: this is the user facing OpenID4VP flow with a "vp_token" responseType, the demo uses the "vp_token id_token" responseType
-		clientId := requestObject.get(oauth.ClientIDParam)
-		if strings.HasPrefix(clientId, "did:web:") {
-			// client is a cloud wallet with user
-			return r.handleAuthorizeRequestFromHolder(ctx, ownDID, requestObject)
-		} else {
-			return nil, oauth.OAuth2Error{
-				Code:        oauth.InvalidRequest,
-				Description: "client_id must be a did:web",
-			}
-		}
+		// if client_id is a url, we can use OpenID federation for automatic client registration
```

Comment:
?

Reply:
additional comment on how to resolve client based on client_id. This snippet refers to openID federation